sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -254,3 +271,197 @@ def _predict_interval(self, fh, X, coverage):
+
+    StatsForecastBackAdapter is a wrapper for sktime forecasters to be used in
+    StatsForecast composite models.
+
+    Parameters
+    ----------
+    estimator : sktime estimator
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsforecast import StatsForecastMSTL
+    >>> from sktime.forecasting.ets import AutoETS
+
+    >>> y = load_airline()
+    >>> trend_forecaster = AutoETS() # doctest: +SKIP
+    >>> model = StatsForecastMSTL( # doctest: +SKIP
+            season_length=[3,12],
+            trend_forecaster=trend_forecaster
+        )
+    >>> fitted_model = model.fit(y=y) # doctest: +SKIP
+    >>> y_pred = fitted_model.predict(fh=[1,2,3]) # doctest: +SKIP
+    """
+
+    _tags = {
+        "python_dependencies": ["statsforecast", "statsmodels"],
+    }
+
+    def __init__(self, estimator):
+        self.estimator = estimator
+        self.in_sample_fh = None
+        self.in_sample_X = None
+
+        super().__init__()
+
+    def __repr__(self):
+        return "StatsForecastBackAdapter"
+
+    def fit(self, y, X=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : ndarray
+            Clean time series of shape (t, ).
+        X : typing.Optional[numpy.ndarray], default=None
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        from sklearn.base import clone
+
+        self.in_sample_fh = np.flip(-np.arange(len(y)))
+        estimator = clone(self.estimator)
+        clone_self = self.new()
+        clone_self.estimator = estimator.fit(y=y, X=X)
+
+        return clone_self
+
+    def predict(self, h, X=None, level=None):
+        """Make forecasts.
+
+        Parameters
+        ----------
+        h : int
+            Forecast horizon.
+        X : typing.Optional[numpy.ndarray], default=None
+            Optional exogenous of shape (h, n_x).
+        level : typing.Optional[typing.Tuple[int]], default=None
+            Confidence levels (0-100) for prediction intervals.
+
+        Returns
+        -------
+        y_pred : dict
+            Dictionary with entries mean for point predictions and level_* for
+            probabilistic predictions.
+        """
+        mean = self.estimator.predict(fh=range(1, h + 1), X=X)[:, 0]
+        if level is None:
+            return {"mean": mean}
+
+        level = sorted(level)
+        coverage = [round(1 - (_l / 100), 2) for _l in level]
+
+        pred_int = self.estimator.predict_interval(
+            fh=range(1, h + 1), X=X, coverage=coverage
+        )
+
+        return {
+            "mean": mean,
+            **{
+                f"lo-{_l}": pred_int[("Coverage", c, "lower")].values
+                for c, _l in zip(reversed(coverage), reversed(level))
+            },
+            **{
+                f"hi-{_l}": pred_int[("Coverage", c, "upper")].values
+                for c, _l in zip(coverage, level)
+            },
```

Comment:
So, I believe this part works as of now only for univariate data? As far as I know, `Coverage` will be present only in case of univairate, and not if there are multiple targets. Have you tried passing multivariate data? How does it work then?

Also, there is a PR which changes this "Coverage" behaviour soon. So, it's probably better to change to at least that format.

Reply:
Internally, `statsforecast` MSTL doesn't support multivariate data, it simply takes first column if it's given multivariate data. Also, what PR are you referencing for the updated behavior of `Coverage`? If it's #4880, I think this won't be a problem, as multivariate forecasts are not supported?